sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		#generate the array
+		roomlist = []
+		data.each do |room|
+		  roomlist.push(room.id)
+		end
+
+		#return the full list of rooms
+		return roomlist
+	end #region_room_list
+	
+	#writes a roomid to the nogo list which is subtracted from the walk_list when the script is run in the future
+	def add_to_nogo(banroom, file_data)
+		#initialize a new array if nogo is empty
+		if file_data.SigilInfo['nogo'] == nil
+		  file_data.SigilInfo['nogo'] = []
+		end
+
+		#push the roomid to the array and remove any duplicates
+		file_data.SigilInfo['nogo'].push(banroom)
+		file_data.SigilInfo['nogo'].uniq
+
+		#alert the player which room has been added to the nogo list
+		DRC.message("Unsuitable room for sigil harvesting. Adding Room #{banroom} to no-go list.")
+		File.open("scripts/data/base-sigils.yaml", "w") { |file| file.write(file_data.to_yaml) } #write roomid to file
+		Flags.reset('sigil-danger-room')
+		pause 0.2
+	end #add_to_nogo
+	
+	#reads base hunting data and combines all rooms with rooms from nogo into a single array
+	def generate_nogo_list(room_list)
+		DRC.message("Pruning walk list to avoid combat. Current list size: #{room_list.length}")
+		ban_list = get_data('hunting').hunting_zones.values.flatten.uniq
+		ban_list += @data.SigilInfo['nogo']
+
+		#subtract array of potentially dangerous rooms from the room list that was passed in
+		room_list -= ban_list
+		DRC.message("Walk list has been pruned. New list size: #{room_list.length}")
+		pause 1
+		return room_list #pruned, safer roomlist is returned
+	end #generate_nogo_list
+
+	#remove any rooms in the nogo list from recorded sigil locations before the script exits
+	def prune_final_list
+		ban_list = get_data('hunting').hunting_zones.values.flatten.uniq
+		ban_list += @data.SigilInfo['nogo']
+
+		@city_list.each do |city|
+			@sigil_list.each do |sigil|
+				@season_list.each do |season|
+					@data.SigilInfo[city][sigil][season] -= ban_list if @data.SigilInfo[city][sigil][season] != nil
+				end
+			end
+		end
+	end #prune_final_list
+	
+	#writes the rooms sigil data to base-sigils.yaml
+	def record_sigil(found_sigils, file_data)
+
+		#check to make sure the sigil array passed in is not empty
+		DRC.message("Writing following data to base-sigils: #{found_sigils}") unless found_sigils.length == 0
+
+		#loop through the sigils and place them in the correct spot in @data
+		found_sigils.each do |sigil|
+		  #if the array doesn't exist yet, initialize a new one
+		  if file_data.SigilInfo[@region][sigil][@season] == nil
+		    file_data.SigilInfo[@region][sigil][@season] = []
+		  end
+		  #push the current roomid into the hash/array under the correct region, sigil, and season keys
+		  file_data.SigilInfo[@region][sigil][@season].push(Room.current.id)
+		  file_data.SigilInfo[@region][sigil][@season].to_set
```

Comment:
🤔 I don't think calling `to_set` transforms the variable it's called on in-place. You may need to do a variable re-assignment. What's your goal with this line?

Reply:
I was trying another way of doing .uniq

I am moving this to 'unless' and 'include?' method as well.